src/tools/tidy/src/rustdoc_html_ids.rs

```diff
@@ -0,0 +1,164 @@
+//! Checks that the rustdoc ID map is up-to-date. The goal here is to check a few things:
+//!
+//! * All IDs created by rustdoc (through JS or files generation) are declared in the ID map.
+//! * There are no unused IDs.
+
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::Path;
+
+use regex::Regex;
+
+const ID_MAP_PATH: &str = "librustdoc/html/markdown.rs";
```

Comment:
I think this will be need to be updated once we only check `src/librustdoc`.

```suggestion
const ID_MAP_PATH: &str = "html/markdown.rs";
```

Reply:
That's what I said above, I'm not sure it's worth it to update it.